components/scream/src/share/mpi/scream_comm.hpp

```diff
@@ -0,0 +1,51 @@
+#ifndef SCREAM_COMM_HPP
+#define SCREAM_COMM_HPP
+
+#include <mpi.h>
+
+namespace scream
+{
+
+// A small wrapper around an MPI_Comm, together with its rank/size
+
+// NOTE: this class checks that MPI is already init-ed, and errors out
+//       if it is not. It is YOUR responsibility to make sure MPI is
+//       init-ed before you create a scream::Comm
+
+class Comm
+{
+public:
+
+  // The default comm creates a wrapper to MPI_COMM_SELF, rather than MPI_COMM_WORLD,
+  // because it is safer to assume I'm the only proc in the comm rather than assuming
+  // that the whole world is in my group.
+  Comm ();
+
+  // This constructor wraps the given MPI_Comm
+  Comm (MPI_Comm mpi_comm);
+
+  // This method resets the stored MPI_Comm to the given one,
+  // updating m_size and m_rank accordingly.
+  // WARNING: it is YOUR responsibility to ensure that
+  //   1) MPI is already inited (call check_mpi_init first, if you are not sure)
+  //   2) the call is collective on both the stored and input comm's
+  void reset_mpi_comm (MPI_Comm new_mpi_comm);
+
+  bool root () const { return m_rank==0; }
```

Comment:
I think we want root() to return the root rank and am_root to return rank == root().

Reply:
Well, isn't the root rank always 0? But yes, the check method can be more verbose, like `am_root`.